src/core/credentials/transport/tls/spiffe_utils.h

```diff
@@ -45,6 +47,99 @@ class SpiffeId final {
   const std::string path_;
 };
 
+// An entry in the Key vector of a SPIFFE Bundle following these documents:
+// https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Trust_Domain_and_Bundle.md#3-spiffe-bundles
+// https://github.com/grpc/proposal/blob/master/A87-mtls-spiffe-support.md
+class SpiffeBundleKey final {
+ public:
+  static const JsonLoaderInterface* JsonLoader(const JsonArgs&);
+  void JsonPostLoad(const Json& json, const JsonArgs&,
+                    ValidationErrors* errors);
+
+  // Returns the PEM x509 string for this SPIFFE Bundle entry which is the only
+  // entry in the x5c_ vector.
+  absl::StatusOr<absl::string_view> GetRoot();
+
+ private:
+  // The following are all fields of the the JWK key in the SPIFFE bundle per
+  // https://github.com/spiffe/spiffe/blob/main/standards/SPIFFE_Trust_Domain_and_Bundle.md#3-spiffe-bundles
+  std::string kid_;
```

Comment:
It doesn't look like we're actually using the `kid`, `n`, or `e` fields anywhere.  Do we actually need them?

Reply:
Good point - I was trying to be as exact with the spec as possible, but we are using a specific subsection of it, so I think we can remove these and add them later if needed.